dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ResourcesServiceImpl.java

```diff
@@ -651,7 +666,14 @@ private boolean upload(User loginUser, String fullName, MultipartFile file, Reso
             return false;
         }
         // query tenant
-        String tenantCode = tenantMapper.queryById(loginUser.getTenantId()).getTenantCode();
+        Tenant tenant = tenantMapper.queryById(loginUser.getTenantId());
+        // this could happen when user logs in as admin
+        if (tenant == null) {
+            logger.error("fail to acquire tenant, please set tenant for login user/admin first");
```

Comment:
It's better to put this message to the result's ```msg``` to improve user friendly.

Reply:
Sure

Reply:
@SbloodyS Just for double-check, I've already put some instructions and users will get it on Web UI once triggered. Where else exactly should I put the message you mentioned above?

![image](https://user-images.githubusercontent.com/34905992/171079333-798daa59-434c-4aee-8ced-17f97c4b29e7.png)

Reply:
What i mean is the ```logger.error``` should put to WEB UI.

Reply:
Looks like `CURRENT_LOGIN_USER_TENANT_NOT_EXIST` already has instructive message here. Do you have any suggestions on how to put `logger.error` to WEB UI? We also need to think about i18n issue. 
![image](https://user-images.githubusercontent.com/34905992/171081106-cbd6f2c6-26e4-4013-bca8-9976574ac1e3.png)

Reply:
Are we still need this? in this patch we already have function verifyTenant in and function upload only working on createResource and updateResource. If we add verifyTenant to both of them, do we still need this check in upload?

Reply:
> Are we still need this? in this patch we already have function `verifyTenant` in and function upload only working on `createResource` and `updateResource`. If we add `verifyTenant` to both of them, do we still need this check in `upload`?

As long as the `verifyTenant` method added to both `createResource` and `updateResource`,  we no more need checks in `upload` method.

Reply:
Yeah, that what I mean, I think we could remove the `Tenant tenant = tenantMapper.queryById(loginUser.getTenantId());` in L669 for this reason

Reply:
@zhongjiajie Looks like we only need to `verifyTenant` in `createResource`. `updateResource` has already done it.
https://github.com/apache/dolphinscheduler/blob/a8a5da367ad37911ee2400df894cdf8ff16fb73e/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ResourcesServiceImpl.java#L400-L404
https://github.com/apache/dolphinscheduler/blob/a8a5da367ad37911ee2400df894cdf8ff16fb73e/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ResourcesServiceImpl.java#L1515-L1530

Reply:
It is a little odd here, since we already have some check func `getTenantCode`, is there a reason we should add new new on named `verifyTenant` to do that?

Reply:
> It is a little odd here, since we already have some check func `getTenantCode`, is there a reason we should add new new on named `verifyTenant` to do that?

Honestly, I just found there is such a `getTenantCode` check. :rofl: What about remove the new `verifyTenant` and add `getTenantCode` in `createResource` to keep consistency? BTW we can keep the unit test to add some coverage.

Reply:
I think is OK, and is good for the
> we can keep the unit test to add some coverage.